components/collector/src/source_collectors/gitlab/failed_jobs.py

```diff
@@ -1,17 +1,13 @@
 """GitLab failed jobs collector."""
 
-from collector_utilities.type import URL, Job
+from collector_utilities.type import Job
 
 from .base import GitLabJobsBase
 
 
 class GitLabFailedJobs(GitLabJobsBase):
     """Collector class to get failed job counts from GitLab."""
 
-    async def _api_url(self) -> URL:
```

Comment:
Don't we need to replace this by a filter function to send only the failed jobs to the server? If I understand all of this correctly, now we changed 'failed jobs' to 'all jobs'

Reply:
Of course, for that we would need some way to identify jobs over multiple 'runs'. Don't know if that is possible with the gitlab api.

Reply:
The filter function is part of the subclass:

```python
class GitLabFailedJobs(GitLabJobsBase):
    """Collector class to get failed job counts from GitLab."""

    def _count_job(self, job: Job) -> bool:
        """Return whether the job has failed."""
        failure_types = list(self._parameter("failure_type"))
        return super()._count_job(job) and job["status"] in failure_types
```